client/src/www/model/errors.ts

```diff
@@ -47,8 +47,18 @@ export class SessionConfigFetchFailed extends CustomError {
 }
 
 export class SessionConfigError extends CustomError {
-  constructor(message: string) {
+  constructor(message: string, options?: {cause?: Error}) {
+    super(message, options);
+  }
+}
+
+export class SessionProviderError extends CustomError {
+  readonly details: string | undefined;
+
+  constructor(message: string, options?: {details?: string}) {
```

Comment:
Why not pass the details directly?

Reply:
This doesn't work?
```suggestion
  constructor(message: string, private options?: string) {
```

Anyway, not important.

Reply:
I guess using the object will give us some space to include `cause` in the `options` in the future.